src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -976,66 +975,45 @@ public String toString() {
     private static void cachePut(Tangle tangle, TransactionViewModel transactionViewModel, Hash hash) throws Exception {
         Cache<Indexable, TransactionViewModel> cache = tangle.getCache(TransactionViewModel.class);
         if (cache.getSize() >= cache.getConfiguration().getMaxSize()) {
-            cacheRelease(tangle);
+            cacheRelease(tangle, cache);
         }
         cache.put(hash, transactionViewModel);
     }
 
     /**
      * Release all transactions from cache.
-     * 
+     *
      * @param tangle Tangle
      * @throws Exception Exception
      */
     public static void cacheReleaseAll(Tangle tangle) throws Exception {
         Cache<Indexable, TransactionViewModel> cache = tangle.getCache(TransactionViewModel.class);
-        Queue<Indexable> releaseQueueCopy = cache.getReleaseQueueCopy();
-        List<Indexable> hashesToRelease = new ArrayList<>();
-        List<Pair<Indexable, Persistable>> batch = new ArrayList<>();
-
-        while (!releaseQueueCopy.isEmpty()) {
-            Indexable hash = releaseQueueCopy.poll();
-            if (hash != null) {
-                TransactionViewModel tvm = cache.get(hash);
-                hashesToRelease.add(hash);
-                if (tvm != null && tvm.shouldPersist()) {
-                    tvm.setShouldPersist(false);
-                    cache.put(hash, tvm);
-                    batch.addAll(tvm.getSaveBatch());
-                }
-            }
+        Indexable hash = cache.getNextReleaseKey();
+        while(hash != null) {
```

Comment:
This could be replaced with a bool return as described below and save the need to make the extra getNextReleaseKey calls to the cache.